src/test/ui/macros/rfc-3086-metavar-expr/syntax-errors.rs

```diff
@@ -142,6 +142,26 @@ macro_rules! unknown_metavar {
     //~| ERROR expected expression
 }
 
+macro_rules! _super {
+    () => {
+        macro_rules! _sub {
+            () => {{
+                $$crate
+                //~^ ERROR unexpected token: crate
+            }};
+        }
+    };
+}
+
+
+macro_rules! dollar_dollar_crate {
+    ($d:tt) => {
+        const _: usize = $d$d crate::IDX;
+        //~^ ERROR expected expression, found `$`
+    };
+}
+dollar_dollar_crate!($);
```

Comment:
This isn't quite the test I had in mind. Instead,

```suggestion
macro_rules! dollar_dollar_crate {
    ($d:tt) => {
        const _: usize = $d$d crate::IDX;
        //~^ ERROR expected expression, found `$`
    };
}
macro_rules! dollar_dollar_use {
    ($d:tt) => {
        dollar_dollar_crate!($d);
    }
}
dollar_dollar_use!($);
```

The fact that the current error message is "expected expression, found `$`" and not "$$crate is not allowed" makes me worried that this actually will work i.e. `$$crate` is insufficiently gated.

Reply:
Changed as requested